docs/integration-guides/ipc-integration.md

```diff
@@ -0,0 +1,288 @@
+
+For testing IPC without caring about permissions, this gives access to everything:
+
+```toml
+[[user]]
+allow = "unrestricted"
+```
+
+A more elaborate sample:
+
+```toml
+[[role]]
+id = "process_admin"
+allow = "api_service_stop"
+
+[[role]]
+id = "walletadmin"
+allow = "wallet_read, wallet_write"
+
+# Add some additional permissions to the default user
+# This is used when no credentials are given
+[[user]]
+allow = "wallet_seed_change, epoch_upgrade, wallet_read"
+
+# The default user can also be set to bare (no default permissions). 
+# That way, a node can be exposed with a limited set of default permissions.
+#bare = true
+
+[[user]]
+id = "mywalletuser"
+# Inherit all the permissions from these roles
+roles = "process_admin, walletadmin"
+
+# Add additional permissions for this specific user
+allow = "wallet_seed_change, epoch_upgrade"
+
+# Default and inherited permissions can be denied
+deny = "wallet_read"
+
+[[user]]
+id = "epoch-user"
+
+# Add specific permission for this user
+allow = "epoch_upgrade"
+
+# Do not inherit any default permissions. This is useful
+# for making users with an explicit set of minimum permissions.
+bare = true
+```
+
+### Reload config
+
+The access file can be reloaded without restarting the node or wallet. For the node:
+
+`killall -SIGHUP nano_node`
+
+(actual syntax depends on OS)
+
+---
+
+## IPC V1 Details
+
+??? info
+
+	As of v18, the Nano node exposes a low level IPC interface over which multiple future APIs can be marshalled. Currently, the IPC interface supports the legacy RPC JSON format. The HTTP based RPC server is still available. Because the only IPC encoding is currently "legacy RPC", RPC config options like "enable_control" still apply.
+
+	**Transports**
+
+	TCP and unix domain sockets are supported. Named pipes and shared memory may be supported in future releases.
+
+	**IPC clients**
+
+	A demo web server written in Go is available at https://github.com/nanocurrency/rpc-go. This allows HTTP clients to make JSON requests via IPC, which is compatible with the existing format. The web server can communicate with a node over domain sockets or TCP.
+
+	A NodeJS client is available at https://github.com/meltingice/nano-ipc-js
+
+	A Python client is being developed at https://github.com/guilhermelawless/nano-ipc-py
+
+	**Configuration**
+
+	IPC is configured in the `node.ipc.tcp` and `node.ipc.local` sections of the node configuration file. A documented configuration file can be generated by following the instructions in [Configuration](/running-a-node/configuration/#ipc).
+
+	**IPC request/response format**
+
+	A client must make requests using the following framing format:
+
+	```
+	REQUEST  ::= HEADER PAYLOAD
+	HEADER   ::= u8('N') ENCODING u8(0) u8(0)
+	ENCODING ::= u8(1)
+	PAYLOAD  ::= <encoding specific>
+	```
+
+	Two encodings currently exist:
+
+	* 1: legacy RPC [_since v18.0_]
+	* 2: legacy RPC allowing unsafe operations if node is configured so [_since v19.0_]
```

Comment:
* 3: flatbuffers [_since v21.0_]
* 4: json over flatbuffers [_since v21.0_]

Reply:
Maybe it should be mentioned somewhere that the _node_ config for IPC and the encoding formats still apply to IPC 2.0.

Reply:
Added new encodings and moved config/encoding details up top as well. Not sure if that the appropriate approach, so review needed.

Reply:
Top level sounds good - looks like the encodings are duplicated now though.

If the entire `**IPC request/response format**` was changed to `## IPC request/response format`, we should be good, since then it applies to all versions.

The list of encodings under `**Configuration**` can then be removed.

Reply:
I think I got the adjustments right. Feel free to write over top of the branch if not though.